scripts/fetch-latest-parity.js

```diff
@@ -0,0 +1,98 @@
+// https://vanity-service.parity.io/parity-binaries?version=stable&os=linux&architecture=x86_64
+// https://vanity-service.parity.io/parity-binaries?version=beta&os=windows&architecture=x86_64
+// https://vanity-service.parity.io/parity-binaries?version=beta&os=darwin&architecture=x86_64
+
+const crypto = require('crypto');
+const { chmod, existsSync, writeFile } = require('fs');
+const download = require('download');
+const fetch = require('node-fetch');
+const { promisify } = require('util');
+const semver = require('semver');
+
+const {
+  parity: { version: versionRequirement, channel: track }
+} = require('../packages/fether-electron/package.json');
+
+const exec = promisify(require('child_process').exec);
+const fsChmod = promisify(chmod);
+const fsWriteFile = promisify(writeFile);
+
+let os;
+switch (process.platform) {
+  case 'win32':
+    os = 'windows';
+    break;
+  case 'darwin':
+    os = 'darwin';
+    break;
+  default:
+    os = 'linux';
+}
+
+const ENDPOINT = `https://vanity-service.parity.io/parity-binaries?version=${track}&os=${os}&architecture=x86_64`;
+
+const STATIC_DIRECTORY = '../packages/fether-electron/static/';
+
+const foundPath = [
+  `${STATIC_DIRECTORY}/parity`,
+  `${STATIC_DIRECTORY}/parity.exe`
+].find(existsSync);
+
+if (foundPath) {
+  // Bundled Parity was found, we check if the version matches the minimum requirements
+  exec(`${foundPath} --version`)
+    .then(({ stdout, stderr }) => {
+      if (stderr) throw new Error(stderr);
+
+      const version = stdout.match(/v\d+\.\d+\.\d+/)[0];
+      if (!semver.satisfies(version, versionRequirement)) {
+        console.log(
+          'Bundled Parity Ethereum %s is older than required version %s',
+          version,
+          versionRequirement
+        );
+        return downloadParity();
+      }
+    })
+    .catch(e => {
+      console.error(e);
+      process.exit(1);
+    });
+} else {
+  // Bundled Parity wasn't found, we download the latest version
+  downloadParity().catch(e => {
+    console.error(e);
+    process.exit(1);
+  });
+}
+
+function downloadParity () {
+  return fetch(ENDPOINT)
+    .then(r => r.json())
+    .then(resp =>
+      resp[0].files.find(({ name }) => ['parity', 'parity.exe'].includes(name))
+    )
+    .then(({ name, downloadUrl, checksum: expectedChecksum }) => {
+      console.log('Downloading Parity Ethereum... (%s)', downloadUrl);
```

Comment:
I'm looking at the CI now, could you add the version in this log? The downloadUrl shown is `https://releases.parity.io/ethereum/stable/x86_64-unknown-linux-gnu/parity`

Reply:
Good idea. Done. I think I agree with you that we would have more visibility pinpointing a version for the bundling rather than "latest stable/beta" (which is dependent on the moment Fether was packaged).

Maybe we can put in `fether-electron/package.json` the version requirement (versions accepted for a running Parity instance), and in addition to that the exact version that will be downloded for bundling ?

Reply:
Hmm, I just saw the `track` variable is set to `channel` from package.json. Why not remove the `channel`, and just keep one variable in package.json, which is `version`. It'll be the required version by Fether, and will also be the downloaded version.

I suppose the `~2.4.1` or `^2.4.1` syntax won't work in the vanity service URL as-is, but the script should ideally download the latest version of parity matching the `version` field.

Reply:
As far as I can see, downloading the latest version matching the requirement and downloading the latest beta is equivalent in effect, considering the moment when the bundling takes place (CI)